platform/lanl/DarwinPower9XL.cmake

```diff
@@ -0,0 +1,91 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+message(STATUS "Detected Darwin Power9 with IBM XL. Applying customizations.")
+
+if(CMAKE_VERSION VERSION_LESS "3.15")
+    message(
+        FATAL_ERROR
+        "IBM XL is only supported on cmake version 3.15 and newer. You are \
+        running ${CMAKE_VERSION}."
+    )
+endif()
+
+# Everything seems to be configured with Cuda 10.1. We should perhaps just
+# auto-detect this, though, in case the CUDA version is bumped at some point.
+set(XL_CUDA_VERSION "10.1")
+
+if ("16.1.1.7" VERSION_GREATER CMAKE_CXX_COMPILER_VERSION)
+    message(
+        FATAL_ERROR
+        "Your version of IBM XL is not supported on Darwin Power9. You must \
+        use version 16.1.1.7 or newer.")
+endif()
+
+find_program(GCC NAMES g++)
+execute_process(
+    COMMAND ${GCC} --version
+    OUTPUT_VARIABLE GCC_VERSION_OUTPUT
+)
+
+string(
+    REGEX MATCH "g\\+\\+ \\(GCC\\) ([0-9]+\.[0-9]+\.[0-9]+)"
+    DISCARD ${GCC_VERSION_OUTPUT})
+
+set(XL_GCC_VERSION ${CMAKE_MATCH_1})
```

Comment:
Where does CMAKE_MATCH_1 variable come from? Is this auto generated from the REGEX MATCH in the line above?

Reply:
Yeah, that's right.